src/logger.cpp

```diff
@@ -0,0 +1,38 @@
+#include "logger.h"
+
+QString currentDateTime() {
+    QDateTime date = QDateTime::currentDateTime();
+    QString dateTime = date.toString("dd.MM.yyyy hh:mm:ss");
+    return dateTime;
+}
+
+void Logger::log_error(QString msg) {
+    if(this->output_ == 0) {
+        qInfo() << currentDateTime() << this << ":" << msg; //how to print the class which is being logged
```

Comment:
> QObject::sender() will return you a pointer to signal sender when you call it from the slot.

Just be sure to check for null ptr before doing anything with the result. 
[Source](https://doc.qt.io/qt-5/signalsandslots.html#advanced-signals-and-slots-usage)